lib/scaffold/start.js

```diff
@@ -5,15 +5,114 @@ var BitcoreNode = require('../node');
 var index = require('../');
 var bitcore = require('bitcore-lib');
 var _ = bitcore.deps._;
-var $ = bitcore.util.preconditions;
 var log = index.log;
-var child_process = require('child_process');
-var fs = require('fs');
 var shuttingDown = false;
 
 log.debug = function() {};
 
 /**
+ * This function will instantiate and start a Node, requiring the necessary service
+ * modules, and registering event handlers.
+ * @param {Object} options
+ * @param {Object} options.servicesPath - The path to the location of service modules
+ * @param {String} options.path - The absolute path of the configuration file
+ * @param {Object} options.config - The parsed bitcore-node.json configuration file
+ * @param {Array}  options.config.services - An array of services names.
+ * @param {Object} options.config.servicesConfig - Parameters to pass to each service
+ * @param {String} options.config.datadir - A relative (to options.path) or absolute path to the datadir
+ * @param {String} options.config.network - 'livenet', 'testnet' or 'regtest
+ * @param {Number} options.config.port - The port to use for the web service
+ */
+function start(options) {
+  /* jshint maxstatements: 20 */
+
+  var fullConfig = _.clone(options.config);
+
+  var servicesPath;
+  if (options.servicesPath) {
+    servicesPath = options.servicesPath; // services are in a different directory than the config
+  } else {
+    servicesPath = options.path; // defaults to the same directory
+  }
+
+  fullConfig.path = path.resolve(options.path, './bitcore-node.json');
+
+  if (fullConfig.datadir) {
+    throw new TypeError(
+      'Configuration file (' + fullConfig.path + ') is not compatible with this version.' +
+        ' Please see https://github.com/bitpay/bitcore-node/blob/bitcoind/docs/upgrade.md' +
+        ' for upgrade details.'
+    );
+  }
+
+  fullConfig.services = start.setupServices(require, servicesPath, options.config);
+
+  var node = new BitcoreNode(fullConfig);
+
+  // setup handlers for uncaught exceptions and ctrl+c
+  start.registerExitHandlers(process, node);
+
+  node.on('ready', function() {
+    log.info('Bitcore Node ready');
+  });
+
+  node.on('error', function(err) {
+    log.error(err);
+  });
+
+  node.start(function(err) {
+    if(err) {
+      log.error('Failed to start services');
+      if (err.stack) {
+        log.error(err.stack);
+      }
+      start.cleanShutdown(process, node);
+    }
+  });
+
+  return node;
+
+}
+
+/**
+ * Checks a service for the expected methods
+ * @param {Object} service
+ */
+function checkService(service) {
+  // check that the service supports expected methods
+  if (!service.module.prototype ||
+      !service.module.dependencies ||
+      !service.module.prototype.start ||
+      !service.module.prototype.stop) {
+    throw new Error(
+      'Could not load service "' + service.name + '" as it does not support necessary methods.'
```

Comment:
This error message is not entirely accurate. For example, service.module.dependencies is supposed to be an array, not a function.